gc3apps/ivs.uzh/gtree.py

```diff
@@ -0,0 +1,211 @@
+#! /usr/bin/env python
+#
+#   gtree.py -- Front-end script for running an R script
+#   over a defined Data set on a given number of nodes .
+#
+#   Copyright (C) 2014, 2015 GC3, University of Zurich
+#
+#   This program is free software: you can redistribute it and/or
+#   modify
+#   it under the terms of the GNU General Public License as published by
+#   the Free Software Foundation, either version 3 of the License, or
+#   (at your option) any later version.
+#
+#   This program is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#   GNU General Public License for more details.
+#
+#   You should have received a copy of the GNU General Public License
+#   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+
+It uses the generic `gc3libs.cmdline.SessionBasedScript` framework.
+
+See the output of ``gtree.py --help`` for program usage
+instructions.
+
+Input parameters consists of:
+
+...
+
+Options:
+"""
+
+__version__ = 'development version (SVN $Revision$)'
+# summary of user-visible changes
+__changelog__ = """
+  2016-05-11:
+  * Initial version
+"""
+__author__ = 'Adrian Etter <adrian.etter@uzh.ch>'
+__docformat__ = 'reStructuredText'
+
+
+# run script, but allow GC3Pie persistence module to access classes defined here;
+# for details, see: https://github.com/uzh/gc3pie/issues/95
+if __name__ == "__main__":
+    import gtree
+    gtree.GtreeScript().run()
+
+from argparse import ArgumentError
+import os
+from os.path import basename, exists, join, realpath
+import sys
+import time
+import tempfile
+
+import shutil
+# import csv
+
+from pkg_resources import Requirement, resource_filename
+
+import gc3libs
+import gc3libs.exceptions
+from gc3libs import Application, Run, Task
+from gc3libs.cmdline import SessionBasedScript, executable_file
+import gc3libs.utils
+from gc3libs.quantity import Memory, kB, MB, MiB, GB, Duration, hours, minutes, seconds
+from gc3libs.workflow import RetryableTask
+
+DEFAULT_CORES = 4
+DEFAULT_MEMORY = Memory(7,GB)
+DEFAULT_WALLTIME = Duration(200,hours)
+
+## custom application class
+class GtreeApplication(Application):
+    """
+    Custom class to wrap the execution of a R script
+    A convenient wrapper script is provided to facilitate how the command file
+    is executed on the remote end.
+    """
+    application_name = 'gtree'
+
+    def __init__(self, input_function, input_dataSet, **extra_args):
```

Comment:
Please use PEP-8 style rules, incl. identifier names. 

In the case at point, `input_dataSet` -> `input_data_set`.